File: Level2/롤 케이크 자르기.swift
import Foundation

func solution(_ topping: [Int]) -> Int {
    var answer = 0
    var toppingDic: [Int: Int] = [:]
    var bordSet = Set<Int>()

    topping.forEach {
        toppingDic[$0, default: 0] += 1
    }

    topping.forEach {
        toppingDic[$0]! -= 1
        bordSet.insert($0)
        
        if toppingDic[$0]! <= 0 {
            toppingDic.removeValue(forKey: $0)
        }

        if toppingDic.count == bordSet.count {
            answer += 1
        }

    }
    return answer
}

File: Level2/롤 케이크 자르기.py
def solution(topping):
    answer = 0
    bordSet = set()
    bordDic = dict()

    for t in topping:
        if t in bordDic:
            bordDic.update({t: bordDic[t] + 1})
        else:
            bordDic[t] = 1

    for t in topping:
        bordDic.update({t: bordDic[t] - 1})
        bordSet.add(t)

        if bordDic[t] <= 0:
            del bordDic[t]

        if len(bordDic) == len(bordSet):
            answer += 1

    return answer
